### PR TITLE
Switch compileViaYul to also where appropriate

### PR DESCRIPTION
**Pull request by axic**
A lot of these tests were set to `true` (aka IR-only), which made no sense if it behaves the same on the old codegen.

### CODE REVIEW

**Comment by axic** [comment]
While all these passed locally, they do not on CI, probably on certain EVM version?

This one I think is because `internal1` and `internal2` are deduplicated in the old codegen but not in the new. Will address:
```
comparison_functions - semanticTests.viaYul
failure

ASSERTION FAILURE:
- file   : boostTest.cpp
- line   : 114
- message: Test expectation mismatch.
Expected result:
// equal() -> true, false, false
// unequal() -> false, true, true

Obtained result:
// equal() -> true, true, false
Warning: The call to "equal()" returned 
[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]
[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]
[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
// unequal() -> false, false, true
Warning: The call to "unequal()" returned 
[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]
```

This seems to have some trailing dirty bytes in the old codegen:
```
function_entry_checks - semanticTests.revertStrings
failure

ASSERTION FAILURE:
- file   : boostTest.cpp
- line   : 114
- message: Test expectation mismatch.
Expected result:
// t(uint256) -> FAILURE, hex"08c379a0", 0x20, 0x12, "Calldata too short"

Obtained result:
// t(uint256) -> FAILURE, 3963877391197344453575983046348115674221700746820753546331534351508065746944, 862718293348820473429344482784628181556388621521298319395315527974912, 0x22414249206465636f64696e673a207475706c65206461746120746f6f, 14677919353341505780581197020104924843037560869400443523087437518186366369792, 0x4100000000000000108866ef2e5600000100000001000000
Warning: The call to "t(uint256)" returned 
[8,c3,79,a0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
[0,0,0,20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
[0,0,0,22,41,42,49,20,64,65,63,6f,64,69,6e,67,3a,20,74,75,70,6c,65,20,64,61,74,61,20,74,6f,6f]
[20,73,68,6f,72,74,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
[0,0,0,0,0,0,0,0,21,0,0,0,0,0,0,0,50,a3,61,f0,2e,56,0,0,c2,97,eb,83,8,2,fd,8e]

Attention: Updates on the test will apply the detected format displayed.
```

File: test/libsolidity/semanticTests/viaYul/comparison_functions.sol
Comment (hrkrshnn) [inline test/libsolidity/semanticTests/viaYul/comparison_functions.sol:29]: I think the test will pass if you remove this line.

Comment (hrkrshnn) [reply test/libsolidity/semanticTests/viaYul/comparison_functions.sol:29]: Had a similar issue in breaking.

**Comment by axic** [comment]
I think these two issues should be investigated, I can take it out to a new PR just to get this large one merged.

**Comment by axic** [comment]
Reverted the two problematic ones, lets merge this and discuss those two separately.

**Review by mijovic — APPROVED** [review]
Looks good, maybe just wait for tests to pass before merge.

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**
